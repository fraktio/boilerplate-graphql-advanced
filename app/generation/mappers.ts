import { CompanyTable } from "~/database/company/companyQueries";
import { PersonTable } from "~/database/person/personQueries";
import { UserTable } from "~/database/user/userQueries";
import { Adult, Company, Person, User } from "~/generation/generated";

export interface UUID extends String {
  _UUID: never;
}

export type UserModel = User & Pick<UserTable, "id">;

export type CompanyModel = Omit<Company, "employees"> &
  Pick<CompanyTable, "id">;

export type PersonModel = Omit<Person, "languages"> & Pick<PersonTable, "id">;

export type AdultModel = Omit<Adult, "languages" | "employers"> &
  Pick<PersonTable, "id">;
