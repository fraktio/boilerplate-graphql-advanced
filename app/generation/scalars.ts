import { Maybe } from "graphql/jsutils/Maybe";
import { CountryCode as BaseCountryCode } from "libphonenumber-js";
import { DateTime } from "luxon";

export interface UUID extends String {
  _id: never;
}

export interface EmailAddress extends String {
  _email: never;
}

export interface FinnishPersonalIdentityCode extends String {
  _personalIdentityCode: never;
}

// eslint-disable-next-line no-shadow
export enum Language {
  EN = "EN",
  FI = "FI",
  SE = "SE",
}

export type CountryCode = BaseCountryCode;

export interface FinnishPostalCode extends String {
  _postalCode: never;
}

export interface FinnishMunicipality extends String {
  _municipality: never;
}

export type Timestamp = Readonly<{
  createdAt: DateTime;
  modifiedAt: Maybe<DateTime>;
}>;
