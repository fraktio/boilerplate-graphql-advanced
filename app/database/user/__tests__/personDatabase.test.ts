import {
  formatUserRow,
  UserID,
  UserTable,
  UserTableRow,
} from "../userDatabase";

import { UUID } from "~/generation/mappers";

const datetimeString =
  "Fri Mar 12 2021 13:16:56 GMT+0200 (Eastern European Standard Time)";

export const dbPersonMockTableRow: UserTableRow = {
  id: (2 as unknown) as UserID,
  uuid: ("valid-uuid" as unknown) as UUID,
  username: "username",
  email: "email",
  phoneNumber: "+358400000000",
  hashedPassword: "valid-hashed-password",
  createdAt: new Date(datetimeString),
  updatedAt: null,
};

export const dbPersonMockTableRowUpdated: UserTableRow = {
  ...dbPersonMockTableRow,
  updatedAt: new Date(datetimeString),
};

export const createFakeUserTable = (person = dbPersonMockTableRow): UserTable =>
  formatUserRow(person);

describe("database tests / user", () => {
  it("formatUserRow", async () => {
    const result = formatUserRow(dbPersonMockTableRow);

    expect(result).toMatchSnapshot();
  });

  it("formatUserRow updated", async () => {
    const result = formatUserRow(dbPersonMockTableRowUpdated);

    expect(result).toMatchSnapshot();
  });
});
