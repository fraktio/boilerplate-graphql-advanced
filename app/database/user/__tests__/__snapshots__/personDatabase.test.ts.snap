// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`database tests / user formatUserRow 1`] = `
Object {
  "UUID": "valid-uuid",
  "email": "email",
  "hashedPassword": "valid-hashed-password",
  "id": 2,
  "phoneNumber": "+358400000000",
  "timestamp": Object {
    "createdAt": "2021-03-12T11:16:56.000+00:00",
    "updatedAt": null,
  },
  "username": "username",
}
`;

exports[`database tests / user formatUserRow updated 1`] = `
Object {
  "UUID": "valid-uuid",
  "email": "email",
  "hashedPassword": "valid-hashed-password",
  "id": 2,
  "phoneNumber": "+358400000000",
  "timestamp": Object {
    "createdAt": "2021-03-12T11:16:56.000+00:00",
    "updatedAt": "2021-03-12T11:16:56.000+00:00",
  },
  "username": "username",
}
`;
